swap_meet/vendor.py

```diff
@@ -1,2 +1,81 @@
 class Vendor:
-    pass
+    def __init__(self, inventory = None):
+        if inventory:
+            self.inventory = inventory
+        else:
+            self.inventory = []
+
+    def add(self, item):
+        self.item = item
+        self.inventory.append(item)
+        return self.item
+
+    def remove(self, item):
+        self.item = item
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_category(self, category):
+        items_in_inventory = []
+        self.category = category
+        for self.item in self.inventory:
+            if self.item.category == self.category:
+                items_in_inventory.append(self.item)
+        return items_in_inventory
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        """Swaps an item from vendor inventory (self.inventory)
+        with an item from other vendor inventory (other.inventory)"""
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        else:
+            self.inventory.remove(my_item)
+            other_vendor.inventory.append(my_item)
+            other_vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+        return True
```

Comment:
Nice guard clause! We can actually omit the else clause here as the function will execute the same without it.

```suggestion
    def swap_items(self, other_vendor, my_item, their_item):
        """Swaps an item from vendor inventory (self.inventory)
        with an item from other vendor inventory (other.inventory)"""
        if my_item not in self.inventory or their_item not in other_vendor.inventory:
            return False

        self.inventory.remove(my_item)
        other_vendor.inventory.append(my_item)
        other_vendor.inventory.remove(their_item)
        self.inventory.append(their_item)
        return True
```